python/morsePersistence.py

```diff
@@ -0,0 +1,120 @@
+tTKScalarFieldSmoother1.MaskField = ['POINTS', '']
+
+# create a new 'Calculator'
+sine = Calculator(Input=tTKScalarFieldSmoother1)
+sine.ResultArrayName = 'Sine'
+sine.Function = 'sin(20*coordsX+1.5)+sin(20*coordsY+1.5)'
+
+# create a new 'Calculator'
+distanceField = Calculator(Input=sine)
+distanceField.ResultArrayName = 'DistanceField'
+distanceField.Function = '-sqrt(coordsX*coordsX+coordsY*coordsY)'
+
+# create a new 'Calculator'
+calculator1 = Calculator(Input=distanceField)
+calculator1.ResultArrayName = 'Blend'
+calculator1.Function = 'Sine+5*DistanceField+5*RandomPointScalars'
+
+# create a new 'Extract Surface'
+extractSurface6 = ExtractSurface(Input=calculator1)
+
+# create a new 'Warp By Scalar'
+warpByScalar1 = WarpByScalar(Input=extractSurface6)
+warpByScalar1.Scalars = ['POINTS', 'Blend']
+warpByScalar1.ScaleFactor = 0.05
+
+# create a new 'TTK PersistenceCurve'
+tTKPersistenceCurve1 = TTKPersistenceCurve(Input=warpByScalar1)
+tTKPersistenceCurve1.ScalarField = ['POINTS', 'Blend']
+tTKPersistenceCurve1.InputOffsetField = ['POINTS', '']
+
+# create a new 'TTK PersistenceDiagram'
+tTKPersistenceDiagram1 = TTKPersistenceDiagram(Input=warpByScalar1)
+tTKPersistenceDiagram1.ScalarField = ['POINTS', 'Blend']
+tTKPersistenceDiagram1.InputOffsetField = ['POINTS', '']
+
+# create a new 'Threshold'
+threshold2 = Threshold(Input=tTKPersistenceDiagram1)
+threshold2.Scalars = ['CELLS', 'PairIdentifier']
+threshold2.ThresholdRange = [-1.0, -0.1]
+
+# create a new 'Threshold'
+threshold1 = Threshold(Input=tTKPersistenceDiagram1)
+threshold1.Scalars = ['CELLS', 'PairIdentifier']
+threshold1.ThresholdRange = [0.0, 100000.0]
+
+# create a new 'Threshold'
+persistenceThreshold = Threshold(Input=threshold1)
+persistenceThreshold.Scalars = ['CELLS', 'Persistence']
+persistenceThreshold.ThresholdRange = [0.7, 10000.0]
+
+# create a new 'TTK TopologicalSimplification'
+tTKTopologicalSimplification1 = TTKTopologicalSimplification(Domain=warpByScalar1,
+    Constraints=persistenceThreshold)
+tTKTopologicalSimplification1.ScalarField = ['POINTS', 'Blend']
+tTKTopologicalSimplification1.InputOffsetField = ['POINTS', '']
+tTKTopologicalSimplification1.VertexIdentifierField = ['POINTS', '']
+
+# create a new 'TTK MorseSmaleComplex'
+tTKMorseSmaleComplex1 = TTKMorseSmaleComplex(Input=tTKTopologicalSimplification1)
+tTKMorseSmaleComplex1.ScalarField = ['POINTS', 'Blend']
+tTKMorseSmaleComplex1.OffsetField = ['POINTS', 'OutputOffsetScalarField']
+tTKMorseSmaleComplex1.SaddleConnectors = 0
+
+# find source
+tTKMorseSmaleComplex1_1 = FindSource('TTKMorseSmaleComplex1')
+
+# create a new 'Extract Surface'
+extractSurface5 = ExtractSurface(Input=OutputPort(tTKMorseSmaleComplex1_1,1))
+
+# create a new 'Tube'
+tube3 = Tube(Input=extractSurface5)
+tube3.Scalars = ['POINTS', 'CellDimension']
+tube3.Vectors = ['POINTS', '']
+tube3.Radius = 0.005
+
+# create a new 'TTK IcospheresFromPoints'
+tTKIcospheresFromPoints2 = TTKIcospheresFromPoints(Input=tTKMorseSmaleComplex1)
```

Comment:
in batch mode, I'd typically store the points directly instead of a graphical primitive (i.e. a sphere).
to extract the just the points out of the mesh, use `MaskPoints` (see https://topology-tool-kit.github.io/examples/manifoldCheck/ for an example)